### PR TITLE
More distributions

### PR DESCRIPTION
**Pull request by dirmeier**
Hello all,
this PR implements Weibull, BetaProportion, NegativeBinomial, NegativeBinomial2 and ZeroInflatedNegativeBinomial distributions. The BetaProportion is a reparameterization using a mean and a sample size parameter (https://en.wikipedia.org/wiki/Beta_distribution#Mean_and_sample_size) and can be used for regression. The NB2 is an alternative parameterization using mean and dispersion parameters and also useful for regression.

All of these are frequently used in Bayesian stats and it would be nice if you could merge them. Thanks!

Best regards,
Simon

### CODE REVIEW

**Review by fehiepsi — COMMENTED** [review]
Awesome work, @dirmeier! Thanks for your contribution. Overall looks great to me. I just have some opinions on the name of some parameters and some simplifications.

File: numpyro/distributions/conjugate.py
Comment (fehiepsi) [inline numpyro/distributions/conjugate.py:258]: I think that you can use this formula
```
return -(self.total_count * softplus(self.logits) + value * softplus(-self.logits) + _log_beta_1(self.concentration, value))
```

**Comment by dirmeier** [comment]
Ciao @fehiepsi, I think I addressed all of your comments. I am still receiving some errors on the unit tests, such as:

```
FAILED test/test_distributions.py::test_jit_log_likelihood[NegativeBinomialLogits-None-params116] - AssertionError: 
FAILED test/test_distributions.py::test_jit_log_likelihood[NegativeBinomialLogits-None-params118] - AssertionError: 
FAILED test/test_distributions.py::test_jit_log_likelihood[NegativeBinomial2-None-params120] - AssertionError: 
FAILED test/test_distributions.py::test_gof[BetaProportion-<lambda>-params4] - assert 0.0 > 2e-05
FAILED test/test_distributions.py::test_gof[BetaProportion-<lambda>-params5] - assert 0.0 > 2e-05
```

Could you give me a hint what the issues are with these? Thanks!

Best,
Simon

File: numpyro/distributions/discrete.py
Comment (dirmeier) [reply numpyro/distributions/discrete.py]: I needed to move these into `conjugate.py` in order to avoid circular imports. Is that ok?

File: numpyro/distributions/__init__.py
Comment (fehiepsi) [inline numpyro/distributions/__init__.py:10]: It would be nice to have `NegativeBinomial` which dispatches to either of those two classes. Similar to `Bernoulli` dispatching to `BernoulliLogits` and `BernoulliProbs`

File: numpyro/distributions/continuous.py
Comment (dirmeier) [reply numpyro/distributions/continuous.py]: Ah, nice catch. No, I think it's better without. Bringing people to actively think about parameterization is better than default arguments. But your call.

File: numpyro/distributions/discrete.py
Comment (dirmeier) [reply numpyro/distributions/discrete.py]: Cool! Thanks for the hint.

File: numpyro/distributions/discrete.py
Comment (fehiepsi) [reply numpyro/distributions/discrete.py]: Yup, I imagined circular import happens, so I used
```
import numpyro.distributions.conjugate as conjugate
```
but it is good to know that solution doesn't work. Thanks for clarifying!

**Review by fehiepsi — COMMENTED** [review]
Looks great (and clean) to me overall. Could you help me add new entries to [docs](https://github.com/pyro-ppl/numpyro/blob/master/docs/source/distributions.rst)? I'll look at the failing tests.

**Comment by fehiepsi** [comment]
For failing tests, I guess you can do
+ For `test_jit_log_likelihood`, change `atol` from 1e-5 to 2e-5.
+ For gof test, I guess we can increase the value of `concentration`? It seems to me that 10000 samples are not enough for a small `concentration`.

**Review by fehiepsi — DISMISSED** [review]
LGTM pending new entries in the [docs file](https://github.com/pyro-ppl/numpyro/blob/master/docs/source/distributions.rst).

**Comment by dirmeier** [comment]
Ciao @fehiepsi. Yes thanks,  just added them.  The rest is also done I believe. An thanks for the quick reviews.  It's a please to contribute to a project, when the devs are so active :)

**Comment by fehiepsi** [comment]
Thanks, @dirmeier!! My impression through your implementations is you understand numpyro pretty well. I do love to see more contributions from you in the future. :)

**Review by fehiepsi — APPROVED** [review]

**Pull request merged**
